lib/test.js

```diff
@@ -103,6 +106,30 @@ class Test {
   body() {
     return this._body;
   }
+
+  // Private - Validations
+  
+  _initializeBody(body) {
+    this._ensureBodyIsValid(body);
+    this._body = body;
+  }
+  
+  _initializeName(name) {
+    this._ensureNameIsValid(name);
+    this._name = name;
+  }
+  
+  _ensureNameIsValid(name) {
+    if (!isString(name)) {
+      throw 'Test does not have a valid name';
+    }
+  }
+  
+  _ensureBodyIsValid(body) {
+    if (!isFunction(body)) {
```

Comment:
I guess the only thing that needs to change is to allow the `body` to be `undefined` as well, as it is a way to indicate that a test is pending, by no passing that argument:

`test('a WIP test')`